src/Symbolica.Extensions.Configuration.FSharp/Binder.fs

```diff
@@ -2,52 +2,150 @@ namespace Symbolica.Extensions.Configuration.FSharp
 
 open Microsoft.Extensions.Configuration
 
+/// <summary>
+/// A specialised type of Reader monad binding values from an <see cref="IConfiguration" />.
+/// </summary>
+/// <remarks>
+/// By utilising the Reader monad the caller can avoid having to supply an <see cref="IConfiguration" />
+/// until the end of the computation. Making the binding code less cluttered and more declarative.
+/// </remarks>
 type Binder<'a> = Binder of (IConfiguration -> BindResult<'a>)
 
 module Binder =
-    let ofBind b = Binder(fun _ -> b)
-
-    let run (Binder b) = b
-
-    let eval config (b: Binder<'a>) = (run b) config
-
+    /// <summary>Create a <see cref="Binder" /> from a <see cref="BindResult" />.</summary>
+    /// <remarks>
+    /// Effectively this is creating a <see cref="Binder" /> that doesn't need to read from the <see cref="IConfiguration" />.
+    /// </remarks>
+    /// <param name="bindResult">The <see cref="BindResult" /> to create this <see cref="Binder" /> from.</param>
```

Comment:
A "from which" style like you use elsewhere would probably read a bit nicer than the trailing "from".